pandas/core/internals/concat.py

```diff
@@ -471,7 +463,8 @@ def get_reindexed_values(self, empty_dtype: DtypeObj, upcasted_na) -> ArrayLike:
                     if len(values) and values[0] is None:
                         fill_value = None
 
-                if is_datetime64tz_dtype(empty_dtype):
```

Comment:
Technically, these checks would be okay to keep with https://github.com/pandas-dev/pandas/pull/48258 changed?

Reply:
yes

Reply:
I'm fine with waiting on 48258, but also think in general `isinstance(obj.dtype, DatetimeTZDtype)` is a better pattern than `is_datetime64tz_dtype(obj.dtype)` as it is more explicit/performant.  Main downside is having two different ways of doing the same check

Reply:
Okay that sounds like good reasoning to me to move forward with this PR first. I'm not much closer on debugging the seemingly unrelated failures with https://github.com/pandas-dev/pandas/pull/48258 so don't want to block.